minetest/minetest
# Allow for Game-Specific Menu Music

### Goal of the PR

Allow games to provide their own menu music theme ~~, via a file of the same name as the game name in /menu~~. For example, The game labyrinth would have a music file at "/games/labyrinth/menu/theme.ogg"

### Implementation

Old menu music could be provided via a "/sounds/main_menu.ogg" file (or "main_menu.#.ogg") by users if they desired.

That functionality still exists in this implementation, but now in the single-player tab, a game can provide it's own music to be played when the tab is open. If the game does not provide it's own music, no music is played. 

~~To safely stop main menu music when switching to this tab or vice versa, I implemented a function that I thought has been missing from minetest for sometime, and was needed in this case:~~

~~#### sound_stop_all()~~

~~This function stops all currently playing sounds, without a user needing to provide a handle~~

### How to test

Download an example game with a menu music file: [Labyrinth on ContentDB](https://content.minetest.net/packages/Just_Visiting/labyrinth/) and click on that game in the singleplayer tab. You will hear some music playing. To test the main menu music, feel free to copy that music file to minetest_folder/sounds/ and rename it to "main_menu.ogg". I recommend using a different sound file to see the differences of when music plays or doesn't. 

### Relevant Issues

#8596

### Ready for Review.
